overlays/haskell.nix

```diff
@@ -598,17 +614,18 @@ final: prev: {
 
         # Add this to your tests to make all the dependencies of haskell.nix
         # are tested and cached.
-        haskellNixRoots = final.recurseIntoAttrs {
-          Level0 = haskellNixRoots' 0;
-          Level1 = haskellNixRoots' 1;
+        haskellNixRoots = final.haskell-nix.roots final.haskell-nix.defaultCompilerNixName;
```

Comment:
Maybe a specific deprecation message here saying to use `project.roots` instead?

Reply:
Done